app/controllers/avo/associations_controller.rb

```diff
@@ -78,7 +80,13 @@ def create_association
       association_name = BaseResource.valid_association_name(@record, association_from_params)
 
       perform_action_and_record_errors do
-        if has_many_reflection?
+        if through_reflection?
+          if additional_params?
+            new_join_record.save
+          else
+            @record.send(association_name) << @attachment_record
+          end
+        elsif has_many_reflection?
```

Comment:
```suggestion
        if through_reflection? && additional_params?
          new_join_record.save
        elsif has_many_reflection? || through_reflection?
```

Reply:
Let's try to avoid repeated code like `@record.send(association_name) << @attachment_record